test/integration2/schema/foreign-keys.spec.js

```diff
@@ -136,6 +136,32 @@ describe('Schema', () => {
             ]);
           });
 
+          it('generates correct SQL for the new foreign key operation with an on deferred constraint', async () => {
+            if (!(isPostgreSQL(knex) || isOracle())) {
+              return;
+            }
+
+            const builder = knex.schema.alterTable(
+              'foreign_keys_table_one',
+              (table) => {
+                table
+                  .foreign('fkey_three')
+                  .references('foreign_keys_table_three.id')
+                  .deferred();
+              }
+            );
+
+            const queries = await builder.generateDdlCommands();
+            console.log(queries.sql);
+            expect(queries.sql).to.eql([
```

Comment:
Unit tests are sufficient to test what SQL is generated in this case, can you also execute the query to ensure that it is actually working? It would be ideal to also replicate scenario where data is eventually consistent, as in row referred to in another row appears within same transaction, but later.